app/scripts/controllers/main.js

```diff
@@ -186,6 +190,42 @@ angular.module('moodCatApp')
           list.scrollTop = list.scrollHeight;
         })
       }
+
+      $scope.setLatestMessage = function(gottenmessages) {
+        var biggest = -1;
+        for (var i = 0; i < gottenmessages.length; i++) {
+            if (gottenmessages[i].id > biggest) {
+             biggest = gottenmessages[i].id;
+            }
+        }
+        $scope.room.lastMessageId = biggest;
+      }
+
+      $scope.firstLatestMessage = function(){
+          $scope.setLatestMessage($scope.messages);
+      }
+
+      $scope.fetchMessages = function(){
+        if(!angular.isObject($scope.room.lastMessageId)){
+          $scope.firstLatestMessage();
+        }
+        if($scope.room.lastMessageId != -1){
+        chatService.fetchMessagesFromMessage($scope.room.id,$scope.room.lastMessageId).then(function(response){
+          $scope.setLatestMessage(response);
+          $scope.messages = $scope.messages.concat(response);
+        });
+      }
+        else{
+          chatService.fetchMessages($scope.room.id).then(function(response){
+          $scope.setLatestMessage(response);
+          $scope.messages = response;
+        });
+      }
+
+        setTimeout($scope.fetchMessages,1000);
```

Comment:
Please use `$interval` instead of `setTimeout` because this will make it more testable. Then obviously there is no need to recall itself in the method body.
```js
var interval = $interval($scope.fetchMessages.bind(this), 1000);
```

Also, the interval should be cancelled when the scope is destroyed, so you dont have 100 intervals running when you switch rooms multiple times.
```js
$scope.$on('$destroy', $interval.cancel.bind($interval, interval));
```